### PR TITLE
Add more resizing handles

### LINKED ISSUE #2985: Allow user to resize blocks from corners
As a user I want an easier way to resize blocks.

AC1: Selecting the corner of a block, should allow the user to resize the block from the corner.

### PR DESCRIPTION
**Pull request by swissspidy**
**To do**

* [x] Consolidate styling for resizing handles (currently split across 2 files)
* [x] Fix styling for resizing handles (some look different on hover)
* [x] Change block position when dragging from top or left (side or corner)
* [x] E2E Tests

Fixes #2985.

### CODE REVIEW

**Comment by swissspidy** [comment]
Current WIP screenshots:

Screenshot 2019-08-13 at 16 30 57
Screenshot 2019-08-13 at 16 30 52

**Comment by swissspidy** [comment]
@miina I think this is a good opportunity to extract the `onResize` callbacks into a bunch of smaller helper functions. Those are currently quite large. Thoughts?

**Comment by miina** [comment]
>  I think this is a good opportunity to extract the onResize callbacks into a bunch of smaller helper functions. Those are currently quite large.

Sounds good to me! It is getting quite large, especially when we'll add more calculations for additional handles.

By "Rotation handles" in the title, do you mean "resizing handles"? Or is the plan to add more rotation handles as well?

**Comment by swissspidy** [comment]
> By "Rotation handles" in the title, do you mean "resizing handles"? Or is the plan to add more rotation handles as well?

Imagine additional rotation handles at every blue dot 😄 Yes, I meant resizing handles.

**Comment by swissspidy** [comment]
> Change block position when dragging from top or left (side or corner)

@miina In case you wanna tackle this one, that would be greatly appreciated 😄 It would probably take me way longer to dig through the logic there.

**Comment by miina** [comment]
I can try 😄

**Comment by miina** [comment]
Note:
Would be good to get #2863 merged first since it reduces the complexity around the Text block by removing the area added for dragging only.

**Comment by swissspidy** [comment]
Alright! That one's on my list for reviewing today anyway 🙂

**Comment by miina** [comment]
@googlebot I consent.

**Comment by miina** [comment]
The corner handles are still not working when the block is rotated, will continue looking into it.

**Comment by miina** [comment]
The corners are working OK now as well, however, now there's a bug with the block still continuing to change the position when moving the mouse even when it's not resizing anymore due to the minimum size limit.

Looking into this now.

**Comment by miina** [comment]
The issue mentioned above is fixed now as well, will add e2e tests and should be done.

Actually, will check also if this can be fixed:
https://github.com/ampproject/amp-wp/blob/cc10a5d322eb61fa8530965cb16d5e6acaaf9b97/assets/src/stories-editor/components/resizable-box/index.js#L214-L219

^ It's a bit more noticeable now with the corner handles. If it seems to remain a mystery then will create a separate issue for it.

File: assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * Internal dependencies
  */
-import { getResizedWidthAndHeight } from '../';
+import { getResizedWidthAndHeight } from '../helpers';
```
Comment (miina) [inline assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js]: Doesn't look like this test is running, at least not locally with `npm run test:js`.

**Comment by miina** [comment]
Just for information that dragging from the resizing handles seems to work as expected in e2e tests, started adding those, will finish tomorrow.

File: assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js]: File ending needs to be `.test.js`

Comment (miina) [reply assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js]: Hmm, none of the `js` tests seem to have `.test.js` ending, only the `e2e` ones.

I just copied the `getBlockPositioning.js` file over from the general `helpers.js` tests, as you did with `getResizedWidthAndHeight.js` file, didn't change the ending.

Will check the config tomorrow.

**Comment by miina** [comment]
E2e tests are added now. The resizing part should be ready for testing/reviewing.

File: assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/stories-editor/components/resizable-box/tests/getResizedWidthAndHeight.js]: The test folder name should be `test`, not `tests`. Will fix.

**Comment by swissspidy** [comment]
@miina Please review this when you get a chance 🙂 
It affects #2992 a bit, so would be nice if this one's in before that.

**Comment by swissspidy** [comment]
Ah, will need to fix the merge conflict first. Doing that tomorrow.

And for #2992 I'll just rebase it on this branch. That seems to work fine.

**Pull request merged**